### PR TITLE
ENH: Raise error writing excel file with a MultiIndexed DataFrame #9794

### LINKED ISSUE #9794: ERR: raise on mutli-index columns as its not implemented in to_excel
Try to the following code, and you will find column labels mismatched with the data in the Excel file :

`cols = pd.MultiIndex.from_tuples([('site',''),('2014','height'),('2014','weight')])`
`df = pd.DataFrame(np.random.randn(10,3), columns=cols)`
`df.to_excel('test.xlsx', index=False)`

I am using pandas v0.16.0.

### PR DESCRIPTION
**Pull request by hsperr**
Raise a NotImplementedError until fixed.

closes #9794

### CODE REVIEW

File: doc/source/whatsnew/v0.16.1.txt [hunk 1]
```diff
@@ -52,6 +52,8 @@ Enhancements
 
 - Allow timedelta string conversion when leading zero is missing from time definition, ie `0:00:00` vs `00:00:00`. (:issue:`9570`)
 
+- Trying to write an excel file now raises error if the DataFrame has a MultiIndex instead of writing a broken Excel file. (:issue:`9794`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.1.txt]: raises a ``NotImplementedError``

File: pandas/core/frame.py [hunk 2]
```diff
@@ -1244,6 +1244,10 @@ def to_excel(self, excel_writer, sheet_name='Sheet1', na_rep='',
         >>> writer.save()
         """
         from pandas.io.excel import ExcelWriter
+        print self.index.nlevels
```
Comment (jreback) [inline pandas/core/frame.py]: pls remove the print

**Comment by jreback** [comment]
can you add a tests that validates this. otherwise looks good.

**Comment by hsperr** [comment]
Actually I just saw there is a some tests that test writing excel with multiindex

https://github.com/pydata/pandas/blob/master/pandas/io/tests/test_excel.py#L895

Referencing:
#5616

ill figure out what the difference is

**Comment by hsperr** [comment]
I added two tests that make sure writing MultiIndex index works and MultiIndex columns raise.
I added the assertRaises to the tests that failed

**Comment by hsperr** [comment]
fixed the backticks in whats new and modified the test for multicolumn/index writing to assert fails only on multicolumn writes

**Comment by jreback** [comment]
@hsperr thanks!

**Pull request merged**